docs/developer/custom-form-component-support.md

```diff
@@ -0,0 +1,45 @@
+```
+  customComponents: "js/custom_components.js"
+```
+In this example Helm will look in the `js` directory relative to the `values.yaml` 
+
+## Render a custom component
+Signaling to the Kubeapps dashboard that you want to render a custom component is pretty straight forward.  Simply add a `customComponent` field to any form parameter defined in the  `values.json.schema` and that will tell the react application to fetch the component from the custom js bundle. An example parameter could look like:
+```
+    "databaseType": {
+      "type": "string",
+      "form": true,
+      "enum": ["mariadb", "postgresql"],
+      "title": "Database Type",
+      "description": "Allowed values: \"mariadb\" and \"postgresql\"",
+      "customComponent": {
+          "type": "radio",
+          "className": "primary-radio"
+      }
+    }
+```
+Note: The `customComponent` field **MUST BE AN OBJECT**. This design decision was made so that developers can pass extra values/properties into their custom components should they require them.
+
+## Updating values with custom components
+Custom form components would be useless without the ability to interact with the YAML state. To do this your custom components should be set up to receive 2 props: `handleBasicFormParamChange` and `param`. `param` is the current json object this is being rendered (denoted by the `customComponent` field) and `handleBasicFormParamChange` which is a function that updates the YAML file. An example of how you use this function can be found in any of the BasicDeploymentForm components such as the [SliderParam](https://github.com/kubeapps/kubeapps/blob/master/dashboard/src/components/DeploymentFormBody/BasicDeploymentForm/SliderParam.tsx#L47-L53).
+```
+  const handleParamChange = (newValue: number) => {
+    handleBasicFormParamChange(param)({
+      currentTarget: {
+        value: newValue,
+      },
+    } as React.FormEvent<HTMLInputElement>);
+  };
+```
+
+## Tips
+To help you get started we provide some examples that you can try [here](https://github.com/kubeapps/kubeapps/blob/master/docs/developer/examples). The three files should give you a good idea about how to start developing and building your own custom components. `CustomComponent.jsx` is a super simple react component that takes the `handleBasicFormParamChange` and `param` props and renders a button that changes the value to 'test'. `CustomComponent.js` is the JavaScript variant of `CustomComponent.jsx` and `CustomComponent.min.js` is a minified js bundle created using [remote-component-starter](https://github.com/Paciolan/remote-component-starter), which is specifically made to help build components that you want to load remotely with the [remote-component](https://github.com/Paciolan/remote-component) tool used by Kubeapps.
```

Comment:
It would be better if we provide here a step-by-step guide of how to work with `remote-component-starter`. At the beginning, I thought that the minified file was just a performance improvement but giving this a try it's a necessary step to be able to load the component. Also, I was able to use the javascript file (CustomComponent.js) but no the file `CustomComponent.jsx`, how do you transform it? is it manual?

Reply:
JSX files cannot be rendering in browsers natively but included that file so that it would be easier to understand the other 2 files.

Reply:
Got it. That's why it'd be useful to be able to have a step-by-step guide to know how to do all this.